### ISSUE TITLE
Issue with snowflake_unsafe_execute

### ISSUE THREAD

**Issue by ramprasads108** [Snowflake-Labs/terraform-provider-snowflake#2491]
### Terraform CLI and Provider Versions

terraform 1.3.6

### Terraform Configuration

```terraform
terraform {
  required_version = ">= 1.3.6"
  required_providers {
    snowflake = {
      source  = "snowflake-labs/snowflake"
      version = ">=0.82.0"
    }
  }
}

resource "snowflake_unsafe_execute" "test" {
  depends_on = [snowflake_schema.schema]
  execute = "SELECT 18"
  revert  = "SELECT 36"
  query  = "SELECT 108"
}


output "unsafe" {
  depends_on = [snowflake_unsafe_execute.test]
  value = snowflake_unsafe_execute.test.query_results
}
It is returning the previous value (96) instead of 108
```

### Expected Behavior

It is returning the previous value (96) instead of 108

### Actual Behavior

It is returning the previous value (96) instead of 108

### Steps to Reproduce

1. `terraform apply`

### How much impact is this issue causing?

Medium

**Comment by ramprasads108** [comment]
added tfe plan output

**Comment by sfc-gh-asawicki** [comment]
Hey @ramprasads108. Thanks for reporting the issue.

I have merged the fix, it will land with the next release.

**Comment by sfc-gh-asawicki** [comment]
Hey @ramprasads108. We have released the fix as part of [v0.87.0](https://github.com/Snowflake-Labs/terraform-provider-snowflake/releases/tag/v0.87.0) release. Please follow the [migration guide](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/v0.87.0/MIGRATION_GUIDE.md#v0860--v0870) during the update. Please confirm that the issue is resolved in the newest version. Thanks!
